File: tier1/javascript/dc3d44eb/handler.js
const { execFileSync } = require('child_process');

function permitted(name) {
  return /^[a-z-]{1,20}$/.test(name);
}

function handle(name) {
  if (!permitted(name)) {
    return '';
  }
  execFileSync('/usr/bin/report', [name]);
  return name;
}

module.exports = { handle };

File: tier1/java/c1d8c829/Handler.java
import java.io.IOException;

public final class Handler {
    private static boolean permitted(String name) {
        return name.matches("^[a-z-]{1,20}$");
    }

    public static String handle(String name) {
        if (!permitted(name)) {
            return "";
        }
        try {
            Runtime.getRuntime().exec(new String[] {"/usr/bin/report", name});
        } catch (IOException e) {
            return "";
        }
        return name;
    }
}
